src/generic_solver.cpp

```diff
@@ -478,31 +481,128 @@ Sort GenericSolver::make_sort(SortKind sk, const SortVec & sorts) const
 
 Sort GenericSolver::make_sort(const DatatypeDecl & d) const
 {
-  throw NotImplementedException("Generic Solvers do not support datatypes");
+  shared_ptr<GenericDatatypeDecl> gdt_decl =
+      static_pointer_cast<GenericDatatypeDecl>(d);
+  string dt_decl_name = gdt_decl->get_name();
+  assert(name_datatype_map->find(dt_decl_name) != name_datatype_map->end());
+  shared_ptr<GenericDatatype> curr_dt = (*name_datatype_map)[dt_decl_name];
+  if (name_sort_map->find(dt_decl_name) == name_sort_map->end())
+  {
+    // Exact functionality of make_genericsort, without the linking
+    // errors (undefined reference when I call a new make_generic_sort).
```

Comment:
huh? :) I don't understand this, but anyway this doesn't look like something that should get into `master`. Maybe it is time to fix this? If it doesn't work out, please let me know what the problem is offline.